simularium/simularium-website
# Feature/information panel

Time estimate or Size
=======
_medium: 10-15_

Problem
=======
Closes #411 

Solution
========
New redux actions/type/constant/state for `selectedAgent`

### `ViewerPanel`
Retrieve agent meta data via the new viewer callback `onFollowObjectChanged` and put into redux as `AgentMetadata`.

### `ModelPanel` -> `SidebarContents`
Get metadata from state and pass it (and UIDisplayData) down to new Metadata Panel.
Refactored both of these components from Classes into Functional Components, trivial because no lifecycle methods.
Small tweaks to html structure and CSS

### `MetadataPanel`
Hopefully it reads easily, is fairly self explanatory.
Use helper functions and manage a few pieces of state/constants to track whether the panel is open and what it should display.

`selectedAgentNotRendered` is truthy only when the agent type is a positive number, greater than the number of agents in UIDisplayData. I don't know if there are any other ways to check for an agent that exists, but has "left the simulation" that don't collide with the null state where the id is -1.




* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open and close the panel.
2. Select/deselect agents, play the sim and verify that data updates

Screenshots (optional):
-----------------------
<img width="753" alt="Screenshot 2024-06-12 at 4 40 08 PM" src="https://github.com/simularium/simularium-website/assets/24981838/53613cea-794c-4ba9-88e5-cb0550466306">
<img width="707" alt="Screenshot 2024-06-12 at 4 40 16 PM" src="https://github.com/simularium/simularium-website/assets/24981838/f7deb2c1-bc57-4a09-81fd-c5c72d4313ac">
<img width="796" alt="Screenshot 2024-06-12 at 4 40 35 PM" src="https://github.com/simularium/simularium-website/assets/24981838/25b9e95e-7d66-49d7-b520-5745771a8cc6">
<img width="809" alt="Screenshot 2024-06-12 at 4 40 26 PM" src="https://github.com/simularium/simularium-website/assets/24981838/723c7606-cf8b-4375-9325-8fc2511b0c01">


Thanks for contributing!
